### PR TITLE
Add functions to reflect Eq and Ord

### PR DESCRIPTION
**Pull request by nfgrusk**
Is this okay? 🙂

### CODE REVIEW

File: src/Data/Reflection/Ord.purs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+instance reflectEq :: (Reifies s (ReifiedOrd a)) => Eq (ReflectedOrd s a) where
+  eq (ReflectedOrd x) (ReflectedOrd y) =
+    reifiedEq (reflect (Proxy :: Proxy s)) x y
+
+instance reflectOrd :: (Reifies s (ReifiedOrd a)) => Ord (ReflectedOrd s a) where
+  compare (ReflectedOrd x) (ReflectedOrd y) =
+    reifiedCompare (reflect (Proxy :: Proxy s)) x y
+
+reifyOrd :: forall a r. (a -> a -> Boolean) -> (a -> a -> Ordering) -> (forall s. (Reifies s (ReifiedOrd a)) => Proxy s -> r) -> r
```
Comment (paf31) [inline src/Data/Reflection/Ord.purs]: What about implementing the `eq` function in terms of the `compare` function here?

**Comment by paf31** [comment]
👍   Definitely, thanks!

File: src/Data/Reflection/Ord.purs (hunk 1 above)
Comment (nfgrusk) [reply src/Data/Reflection/Ord.purs]: Like this?

```purescript
reifyOrd compare f = reify (ReifiedOrd eq compare) f
  where
    eq x y = case compare x y of
      EQ -> true
      _ -> false
```

Comment (paf31) [reply src/Data/Reflection/Ord.purs]: Yes, exactly.

**Comment by paf31** [comment]
Great, thanks!

**Pull request merged**